### PR TITLE
Sort `/pages` alphabetically

### LINKED ISSUE #922: All Pages list is not in alphabetical order
At the bottom of the `All Pages` special page are a few pages which are not in alphabetical order:

image

### PR DESCRIPTION
**Pull request by techwiz24**
Previously, the 'All Pages' view was not sorted alphabetically. We need
to sort on the lowercase version of the page names so that lowercase
names do not end up at the bottom of the list and are instead mixed in
alphabetically, as they should be.

Fixes #922

### CODE REVIEW

File: lib/gollum/app.rb [hunk 1]
```diff
@@ -450,6 +450,7 @@ def wiki_new
       wiki         = Gollum::Wiki.new(settings.gollum_path, wiki_options)
       @results     = wiki.pages
       @results     += wiki.files if settings.wiki_options[:show_all]
+      @results     = @results.sort_by { |p| p.name.downcase } # Sort Results alphabetically, fixes 922
```
Comment (dometto) [inline lib/gollum/app.rb]: I expected that `sort_by!` would be slightly faster, but I'm actually seeing the opposite when benchmarking. So, whatever. :)

**Comment by dometto** [comment]
Thanks for the PR! Would you perhaps have the chance to add a test for this behavior?

**Comment by techwiz24** [comment]
@dometto Sure! However, it seems I'm having trouble setting up a ruby dev environment on my arch install (i'm probably just missing a package. I've never done anything with ruby before, so I have no clue what to try) and waiting on travis-ci to post stuff back seems counter-intuitive. This is what I get when I try to run `bundle exec rake test` as stated in the README:

```
/usr/bin/ruby -I"lib:lib:test:."  "/usr/lib/ruby/2.2.0/rake/rake_test_loader.rb" "test/**/test_*.rb" 
/home/nathan/projects/gollum/test/helper.rb:3:in `require': cannot load such file -- test/unit (LoadError)
	from /home/nathan/projects/gollum/test/helper.rb:3:in `<top (required)>'
	from /home/nathan/projects/gollum/test/test_allow_editing.rb:2:in `require'
	from /home/nathan/projects/gollum/test/test_allow_editing.rb:2:in `<top (required)>'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:10:in `require'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:10:in `block (2 levels) in <main>'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:9:in `each'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:9:in `block in <main>'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:4:in `select'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:4:in `<main>'
rake aborted!
Command failed with status (1): [ruby -I"lib:lib:test:."  "/usr/lib/ruby/2.2.0/rake/rake_test_loader.rb" "test/**/test_*.rb" ]

Tasks: TOP => default => test
(See full trace by running task with --trace)
```

**Comment by bartkamphorst** [comment]
Did you run `bundle install` before trying `bundle exec rake test`?

**Comment by techwiz24** [comment]
@bartkamphorst yes:

```
[nathan@nathan-arch gollum]$ gem uninstall -aIx gollum
[nathan@nathan-arch gollum]$ clear
[nathan@nathan-arch gollum]$ bundle install
Resolving dependencies...
Using rake 10.4.2
Using i18n 0.7.0
Using multi_json 1.10.1
Using activesupport 3.2.21
Using ansi 1.4.3
Using builder 3.2.2
Using charlock_holmes 0.7.3
Using diff-lcs 1.2.5
Using github-markdown 0.6.8
Using posix-spawn 0.3.9
Using github-markup 1.3.1
Using mime-types 1.25.1
Using gitlab-grit 2.7.3
Using gollum-grit_adapter 0.1.2
Using mini_portile 0.6.2
Using nokogiri 1.6.5
Using rouge 1.7.7
Using sanitize 2.1.0
Using stringex 2.5.2
Using gollum-lib 4.0.1
Using mustache 0.99.8
Using rack 1.6.0
Using rack-protection 1.5.3
Using tilt 1.4.1
Using sinatra 1.4.5
Using useragent 0.10.0
Using gollum 3.1.1 from source at .
Using hashie 3.3.2
Using json 1.8.2
Using metaclass 0.0.4
Using minitest 4.7.5
Using powerbar 1.0.11
Using minitest-reporters 0.14.24
Using mocha 1.0.0
Using rack-test 0.6.3
Using shoulda-context 1.2.1
Using shoulda-matchers 2.7.0
Using shoulda 3.5.0
Using thread_safe 0.3.4
Using tzinfo 1.2.2
Using twitter_cldr 2.4.3
Using bundler 1.7.11
Your bundle is complete!
Use `bundle show [gemname]` to see where a bundled gem is installed.
[nathan@nathan-arch gollum]$ bundle exec rake test
/usr/bin/ruby -I"lib:lib:test:."  "/usr/lib/ruby/2.2.0/rake/rake_test_loader.rb" "test/**/test_*.rb" 
/home/nathan/projects/gollum/test/helper.rb:3:in `require': cannot load such file -- test/unit (LoadError)
	from /home/nathan/projects/gollum/test/helper.rb:3:in `<top (required)>'
	from /home/nathan/projects/gollum/test/test_allow_editing.rb:2:in `require'
	from /home/nathan/projects/gollum/test/test_allow_editing.rb:2:in `<top (required)>'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:10:in `require'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:10:in `block (2 levels) in <main>'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:9:in `each'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:9:in `block in <main>'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:4:in `select'
	from /usr/lib/ruby/2.2.0/rake/rake_test_loader.rb:4:in `<main>'
rake aborted!
Command failed with status (1): [ruby -I"lib:lib:test:."  "/usr/lib/ruby/2.2.0/rake/rake_test_loader.rb" "test/**/test_*.rb" ]

Tasks: TOP => test
(See full trace by running task with --trace)
```

**Comment by dometto** [comment]
Strange. `test-unit` is not in the Gemfile because it's supposed to be included with ruby, but maybe it isn't in arch? Possibly you need something like a ruby-dev package? You could also just try to add it to your Gemfile:

`gem 'test-unit' `

...then `bundle install`

**Comment by techwiz24** [comment]
Adding that to the gemfile seems to have fixed it (I didn't install anything else, strange). I'll work on getting a test written!

**Comment by dometto** [comment]
@techwiz24 I see you've added some tests. Is the PR ready to merge like this?

**Comment by techwiz24** [comment]
Yes, it should be all set. The tests I added pass when the changes in this
PR are applied and fail before. Unless you have anything to add, it should
be good to merge.

Nathan Lowe

**Comment by dometto** [comment]
Nothing to add, but if it's not too much trouble, it would be great if you could rebase this into a single commit.

**Comment by techwiz24** [comment]
Done! Glad I could help!

**Comment by dometto** [comment]
Great, merging! Thanks for spotting the problem and providing the fix. :)

**Pull request merged**